packages/js/src/plugins/typescript/plugin.ts

```diff
@@ -622,30 +618,22 @@ function isValidPackageJsonBuildConfig(
     joinPathFragments(projectRoot, 'package.json')
   );
 
-  const rootDir = tsConfig.options.rootDir ?? 'src/';
-  if (!tsConfig.options.rootDir) {
-    console.warn(
-      `The 'rootDir' option is not set in the tsconfig file at ${tsConfigPath}. Assuming 'src/' as the root directory.`
-    );
-  }
-
-  const isPathWithinSrc = (path: string): boolean => {
-    const resolvedRootDir = resolve(workspaceRoot, projectRoot, rootDir);
-    const pathToCheck = resolve(workspaceRoot, projectRoot, path);
-
-    return pathToCheck.startsWith(resolvedRootDir);
+  const isPathSourceFile = (path: string): boolean => {
+    const ext = extname(path);
```

Comment:
Can we not go off of whether or not the path is in the outDIr of the `tsconfig.json` or in the outputs of the build? Rather than the extension?

Reply:
`outDir` is not always available and if it is not the `outDir` would be the same as the source in this scenario we would need to ensure that the file type is not the same as what is inside the source.